### ISSUE TITLE
`sails.hooks.views.render()` : Render view, in non request scope, do not take locale into account

### ISSUE THREAD

**Issue by jberrisch** [balderdashy/sails#3833]
**Sails version**:0.12.4
**Node version**:4.5.0
**NPM version**:2.15.9
**Operating system**:Os X 10.11.6

<hr/>

I try to use a hook (sails-hook-cron) to send emails to customers who have different locales.
In this hook I use '_sails.hooks.views.render_' to compile my email template.
This works quite fine but when I pass the locale as options it is ignored.
Here is a example hook to reproduce this error:

```
module.exports = function(sails) {
    return {
        initialize: function(cb) {
            sails.on('lifted', function() {
                if (sails.hooks.views && sails.hooks.views.render) {
                    sails.hooks.views.render('email/test', {
                        locale: 'fr'
                    }, (err, compiledEmail) => {
                        console.log(compiledEmail);
                    });
                }
            });

            return cb();
        }
    }
};
```

I got into the source code and at the ligne 102 of the file: **node_modules/sails/lib/hooks/views/render.js** the '_req.locale_' is set to '_options.locale_'.

```
var req = {
  headers: {}
};

// Initialize i18n if hook is enabled
if (sails.hooks.i18n) {
  require('i18n').init(req, options, function() {

    // Set the locale if necessary
    if (options.locale) {
      req.locale = options.locale;
    }

    // Render the view
    sails.config.views.engine.fn(absPathToView, options, cb_view);
  });
}
```

But, ... the i18n module invoked at ligne 98 (_require('i18n').init_) overrides the '_options.locale_' value. 
So at ligne 102 it will always be set to the default locale because i18n guesses the value from the url-params / headers / cookies but never from the options.

By adding:

```
req.headers['accept-language'] = (options.locale || sails.hooks.i18n.defaultLocale);
```

after ligne 97 (**render.js**) it works like expected, but I guess this is not very beautiful.

I tried to write a pull request and a test for this issue, but I had some trouble charging the i18n hook for the test because there's no sample configuration...

**Comment by Josebaseba** [comment]
I can confirm that with sails v0.11.5 it's working properly.

**Comment by jberrisch** [comment]
I downgraded my sails (0.12.4) i18n module to version 0.5.0 (dependency of sails 0.11.5) and it works fine.
They (i18n team) changed a lot in how the .init function works...

**Comment by Josebaseba** [comment]
This feature it's really important for my app, so when I find some time I'll try to find a solution and then send a PR. I want to update to sails v0.12 but I need this functionality running!

**Comment by mikermcneil** [comment]
@Josebaseba That'd be awesome!  Talked to @dougwilson a little bit about this recently, but unfortunately it sounds like there's still a ways to go before views support will be ready to extrapolate from Express.

In the mean time, if you're able to get this working, then we can make [`sails.renderView()` (aka sails.hooks.views.render)](https://github.com/balderdashy/sails/blob/master/lib/hooks/views/render.js#L9-L13) officially supported & documented for Sails v1

**Comment by Josebaseba** [comment]
Done 👍  https://github.com/balderdashy/sails/pull/3889

**Comment by jberrisch** [comment]
Wonderful news, thanks a lot!

**Comment by Josebaseba** [comment]
@jberrisch please can you test it? npm install sails@^0.12.10-1

For me it's working as it should, take a look and the we can close this issue. Thanks!

**Comment by mikermcneil** [comment]
@Josebaseba @jberrisch thanks guys!

_(Now published as 0.12.10)_
